pydatastructs/graphs/algorithms.py

```diff
@@ -877,3 +879,73 @@ def _job(graph: Graph, u: str):
     if len(L) != num_vertices:
         raise ValueError("Graph is not acyclic.")
     return L
+    
+def minDistance(graph: Graph,dist,visited,source):
+    min = sys.maxsize
+    min_index=source
+    for v in graph.vertices:
+        if dist[v] < min and visited[v] is False:
+            min = dist[v] 
+            min_index = v 
+    return min_index 
+        
+def dijkstra_algorithm(graph: Graph,start: str):
+    """
+    Finds shortest path distance in the given graph from a given source to all vertex.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph under consideration.
+    start: str
+        The name of the source the node.
+    
+    Returns
+    =======
+
+    Displays Vertex and Distance from Source as a Key value pair
+    
+    Examples
+    ========
+
+    >>> from pydatastructs import Graph, AdjacencyListGraphNode
+    >>> from pydatastructs import shortest_paths
+    >>> V1 = AdjacencyListGraphNode("V1")
+    >>> V2 = AdjacencyListGraphNode("V2")
+    >>> V3 = AdjacencyListGraphNode("V3")
+    >>> G = Graph(V1, V2, V3)
+    >>> G.add_edge('V2', 'V3', 14)
+    >>> G.add_edge('V1', 'V2', 12)
+    >>> dijkstra_algorithm(G,'V1')
+    {'V1': 0, 'V2': 12, 'V3': 26}
+    
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#:~:text=Dijkstra's
+    """
+    V=len(graph.vertices)
+    visited={}
+    dist={}
+    for v in graph.vertices:
+        visited[v]=False
+        if v!=start:
+            dist[v] =sys.maxsize
+    dist[start] = 0
+        
+    for cout in range(V):
+        u = minDistance(graph,dist,visited,start) 
```

Comment:
Here we can use Fibonacci heap(see heaps in this repo). Right Now complexity will be O(V^2). After Fibonacci heap, it should be O(VlogV).

Reply:
In pydatastructs/trees/heaps.py directory,there is no fibonacci heaps implementation,then whether have to remove min_distance function and include implementation of fibonacci heap in pydatastructs/graphs/algorithms.py.
or otherwise i will push this existing one?

Reply:
You can use `BinomialHeap` in that case.

Reply:
priority queue is it k to use in pydatastructs/pydatastructs/miscellaneous_data_structures/queue.py? to find the vertex which has minimum distance

Reply:
or binomial heap is fine to use for finding vertex position which has minimum distance?

Reply:
I would suggest to use 'binomial_heap' implementation of `PriorityQueue`(read its docs).

Reply:
but inserting updated distances in binomial heap for all vertex takes O(V) time,then wats the use for O(logV) in this case?

Reply:
included priorityqueue ,getting output in spyder but not here in CI/CD pipeline